android-sdk/src/main/java/org/sagebionetworks/bridge/android/manager/BridgeManagerProvider.java

```diff
@@ -0,0 +1,52 @@
+package org.sagebionetworks.bridge.android.manager;
+
+import android.content.Context;
+import android.support.annotation.AnyThread;
+import android.support.annotation.NonNull;
+
+import org.sagebionetworks.bridge.android.BridgeConfig;
+import org.sagebionetworks.bridge.android.manager.auth.AuthenticationManager;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+/**
+ * Created by jyliu on 1/20/2017.
+ */
+@AnyThread
+public class BridgeManagerProvider {
```

Comment:
This class is intended be a single point for configuration and set up of dependencies and Managers. Pass in the context once on application start, and BridgeManagerProvider can be called from any thread/context (background service, foreground activity) to get a Manager.

I think it'll make more sense once I add in more code. One benefit of having this centralized source for interacting with Bridge is it makes it easy to inject a dependency, such as a mock for testing purposes. Code from anywhere can call BridgeManagerProvider.getInstance().getXYZManager, and get the mock.

This is a work in progress too, I've made some minor changes in an upcoming PR.